### PR TITLE
Fix culture format of date for Text function and add tests. 

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1105

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 1]
```diff
@@ -344,7 +344,7 @@ internal static bool TryExpandDateTimeExcelFormatSpecifiersToStringValue(IRConte
                 case "'longtime24'":
                 case "'longtime'":
                 case "'longdate'":
-                    result = new StringValue(irContext, dateTime.ToString(ExpandDateTimeFormatSpecifiers(format, culture)));
+                    result = new StringValue(irContext, dateTime.ToString(ExpandDateTimeFormatSpecifiers(format, culture), culture));
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:347]: nit:  can we extra this part to a local (just breaking up the long expression). 
var formatStr = ExpandDateTimeFormatSpecifiers(format, culture);

That also makes it clearer that the ToString() was missing a culture parameter.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 2]
```diff
@@ -234,4 +234,19 @@ Error({Kind:ErrorKind.Div0})
...
+"5:02:03 CH"
+
+>> Text(DateValue("02/27/2023"), DateTimeFormat.LongDate, "vi-VN")
+"27 Tháng Hai 2023"
+
+>> Text(DateTimeValue("02/27/2023"), DateTimeFormat.LongDateTime, "vi-VN")
+"27 Tháng Hai 2023 12:00:00 SA"
+
+>> Text(Boolean("True"), "", "vi-Vn")
+"true"
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:249]: is there ever a case where we don't return the literal string "true"? 
And can we double check that PowerApps returns the same values for the new cases.

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:249]: as offline chat: only different between our and PowerApps are https://github.com/microsoft/Power-Fx/issues/1120.

**Pull request merged**